components/automate-ui/src/app/pages/project/rules/project-rules.component.html

```diff
@@ -112,15 +112,19 @@
               <chef-form-field class="attribute-field">
                 <label>
                   <span class="label">Value</span>
-                  <chef-input ngDefaultControl
-                    formControlName="values"
-                    [value]="condition.controls.values.value"
-                    data-cy="rule-value"
-                  ></chef-input>
+                  <input
+                  chefInput
+                  formControlName="values"
+                  [value]="condition.controls.values.value"
+                  data-cy="rule-value"
+                  />
                 </label>
-                <chef-error
-                  *ngIf="(ruleForm.get('conditions').controls[i].get('values').hasError('required') || ruleForm.get('conditions').controls[i].get('values').hasError('pattern')) && ruleForm.get('conditions').controls[i].get('values').dirty"
-                >Value is required.</chef-error>
+                <div>
+                  <chef-error
+                      *ngIf="( ruleForm.get('conditions').controls[i].get('values').hasError('required') && ruleForm.get('conditions').controls[i].get('values').touched )
+                              || ( ruleForm.get('conditions').controls[i].get('values').hasError('pattern') && ruleForm.get('conditions').controls[i].get('values').dirty )"
```

Comment:
Wow, that's quite an expression! 😁 
Suggest moving that to the code-behind; try to keep expressions in a template very simple as a rule; either a single variable or a simple expression.
Couple reasons:
The template should define presentation; the code-behind should define execution, so one should strive to maintain separation of concerns.
Also, it makes both testing and diagnosis of issues much more accessible.

Reply:
I'm going to go ahead and merge this, and open up an new ticket to explore ways to shorten this, as I don't want this to be held up any longer.  I agree I need to find a way to better destructure this.  So far, i'm stuck on getting past the `controls` piece, but I imagine there is a better way to do this in a lot places across the application.  Thank you for talking this through earlier with me.